ddtrace/ci_visibility/_recorder.py

```diff
@@ -0,0 +1,143 @@
+import json
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+import ddtrace
+from ddtrace import Tracer
+from ddtrace.ci_visibility._filters import TraceCiVisibilityFilter
+from ddtrace.contrib import trace_utils
+from ddtrace.ext import ci
+from ddtrace.ext import test
+from ddtrace.internal import atexit
+from ddtrace.internal import forksafe
+from ddtrace.internal.compat import parse
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.service import Service
+from ddtrace.settings import IntegrationConfig
+
+
+log = get_logger(__name__)
+
+
+def _extract_repository_name_from_url(repository_url):
+    # type: (str) -> str
+    try:
+        return parse.urlparse(repository_url).path.rstrip(".git").rpartition("/")[-1]
```

Comment:
There's some duplication between the [utils](https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/sourcecode/_utils.py) and [constants](https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/ext/git.py) in the sourcecode integration product. Can we consolidate some of these?

Reply:
I think we should definitely address this duplication as a follow up, but this isn't in the scope of this PR.